### DISCUSSION TITLE
State of Plugin System and ability to template sections

### DISCUSSION THREAD

**Discussion by T2brozz** [outline/outline#8038] [Ideas]
Hello there,
I have to Feature Ideas. The first is section Templating, where you can Templatemize Textsnippet and add theme with a command to the Document.
It would also be nice if its recursive (also for document templates) where I can define a Command, which gets executed when inserting it in a document. So if my template(also document template) includes a `/stuff` as a Placeholder than it gets replaced. 

We have currently for other menbers shortnames , like JS for John Smith , it would be nice if I can also type @JS and it links to the Same thing like when I type @John Simth . alternativly a /js which replaces it with @john smith would be okay. 

so I found the issue #2219 where a system to add Plugins was discussed, but I also saw that there is a plugin's directory.  But nothing more . Is it possible to add these features myself without reading the whole Codebase or is it currently not possible.

**Comment by tommoor** [comment]
For plugins there needs to be a "hook" defined, there isn't anything plugin related at the editor level right now. You would need a hook to add eg commands to the editor, I think it's do-able but the system doesn't exist yet.

I do think that adding templates via the slash menu should be possible though, just not through the plugin system

**Reply by T2brozz** [reply]
mhh oki thanks , I changed the discussion Category to ideas. Maybe something link templates of text is a feature :) I'd be happy

**Comment by gabschne** [comment]
I really support the feature request for "template blocks".

Two additional comments/requests:
- Support for including/using them in page templates and nesting them in other template blocks would be very useful, e.g. `{template[templateName]}`. This would allow building templates that are automatically "updated" as soon as one of the included blocks is changed. **Note:** A loop detection should also be added.
- When adding via the slash menu, something like a fuzzy search should be supported in order to find the wanted one even on instances with a large number of template blocks.

**Reply by T2brozz** [reply]
yeah, but updating of (included) blocks should not affect pages where it was used.

**Reply by Lindenk** [reply]
Maybe an option, but I personally would like a template that can handle formatting (such as making tables/etc) separate from the content that goes into it, and have the result updated if the formatting has changed. Otherwise if the formatting for a template changes you'd have to manually update everything that uses it
